### DISCUSSION TITLE
行動規範に書く内容

### DISCUSSION THREAD

**Discussion by msntts** [msntts/PatternLanguageEditor#1] [Ideas]
このリポジトリで活動する際のルールを決めておこうと思います。  
githubが公開している行動規範についての[ガイドはここ](https://opensource.guide/ja/code-of-conduct/#:~:text=%E8%A1%8C%E5%8B%95%E8%A6%8F%E7%AF%84%E3%81%AF%E3%80%81%E3%81%82%E3%81%AA%E3%81%9F%E3%81%AE,%E3%81%93%E3%81%A8%E3%81%AB%E3%82%82%E5%BD%B9%E3%81%AB%E7%AB%8B%E3%81%A1%E3%81%BE%E3%81%99%E3%80%82)。

ひとまず、思いついた「こういうルールあったらいいんじゃない？」というのをここに残していきましょう✊

**Comment by kinakomochigit** [comment]
少し細かいですが、今思いつくものを書いておきました。
## GitHubについて
- [ ] **commitやpushの頻度（一区切りついたらどんどんしていいか？）**
- [ ] **コミットコメントの書き方（最低限何をかくか？）**
- [ ] ブランチの切り方

## コミュニケーションについて
- [ ] **コミュニケーションをとる手段は何か？（Github上、slack、LINEなど）**
- [ ] どのような頻度でコミュニケーションするか（定例会、困ったとき、とりあえず1週間に一回、決まりなしなど）

## 裁量について
- [ ] **どこまで自律的に動いてほしい、動くべきか（勝手にしていいこと、絶対許可をもらってからするべきこと）**
- [ ] 最終決定権はどうするか？（リーダーにお任せ、多数決、とことん話し合いなど）

## 気持ち
- [ ] **熱量はどのくらいか合わせておきたい（期限をいつするか？、1日or1週間どのくらいやるかなど）**
- [ ] **目標（とりあえず動くものをいつまでに作るとか？）**

**Reply by msntts** [reply]
タイトルを行動規範としてしまったので細かいかな？って思ってしまったかもしれないですね🥶
もらった内容は、貢献方法、リポジトリの運用ルールとかもあって、それらは必要なことだと思うのでジャンル分けして残しましょう📝

**Reply by msntts** [reply]
ドラフト書きました  
👉[check it](https://github.com/msntts/PatternLanguageEditor/discussions/1#discussioncomment-4998201)

**Reply by kinakomochigit** [reply]
ありがとうございます確認しました。技術要素や他にもOSSへの心構えなど私がキャッチアップしないといけないことがあることがよくわかりました。できる限り勉強しておきます。もし、おすすめの参考書などあれば提示していただけると幸いです🙏

**Comment by msntts** [comment]
エクストリームプログラミングのエッセンスから
- YAGNI

今回はwebアプリなので、例えば見た目とかこだわりたくなる事もあるけど、見た目に関わるタスクじゃなかったら触れないとか

あとは、OSSなのでコミュニケーションの原則として
- 反応しなくても良い
  - 出来ないときもあるし、反応したく無い内容の時もあるし

**Comment by msntts** [comment]
もらったメッセージから何個かのグループに分けてルールを決めたいと思います。  

# 行動規範
## 開いた環境
本プロジェクトにかかわるやり取りは[discussions](https://github.com/msntts/PatternLanguageEditor/discussions)や[issues](https://github.com/msntts/PatternLanguageEditor/issues)でおこない、だれでも状況がわかる環境を保ちましょう。  

## 議論の推奨
プロジェクト運営中の不明点、技術的な不明点など困りごとやその他判断に迷うことがあったら気軽に[discussions](https://github.com/msntts/PatternLanguageEditor/discussions)に投稿して課題解決に向けた議論を始めましょう。
ただし、常に回答が得られるわけでもないことも受け入れましょう。  

## 常にシンプルに
開発中にアイディアがわいてきたり、見た目が気になったりすることが多々あると思います。その場合は、一度落ち着いてやりたいことをタスクに書き残して今取り組んでいる作業を完了しましょう。 
合言葉はYou Aren't Gonna Need It(YAGNI)

## 漸進的であること
規範やルールは少しずつ改善し、変化を受け入れていきましょう。

**Reply by msntts** [reply]
# プロジェクト管理
## ブランチの管理について
mainブランチは常にローンチ可能なものだけを登録します。mainブランチへの登録は🍣が担当します。  
ブランチはタスク単位で作成し(タスクから機械的にブランチ名つけられるようにするといいですかね)、タスク以外の内容が含まれるのはNGです。  
タスク単位のブランチが完了したらPull Requestを発行してください。レビュー後squashマージにてmainブランチに取り込みます。  
(squashマージで取り込むので、途中にどんなコミットをどれだけしても大丈夫です)
開発中に使用したブランチはブランチが煩雑になるのを防ぐため、マージ後どこかのタイミングで削除します。

**Reply by msntts** [reply]
# ゴール設定
## マイルストン
22年度時点では、Reactでパターンがブラウザ上で見れるところまでをゴールにしたいと思います。  
その後の技術要素としては盛沢山で

- DSL(langium)
- LSP Client/Server
- グラフUI

等を使って機能アップしていく予定です。マイルストンの打ち合わせはGithubに機能がないので最初は別途開催。  
もし今後メンバーが増えたらやり方を考えましょう。
